src/main/scala/beam/agentsim/agents/planning/BeamPlan.scala

```diff
@@ -92,43 +92,37 @@ class BeamPlan extends Plan {
   //////////////////////////////////////////////////////////////////////
   // Beam-Specific methods
   //////////////////////////////////////////////////////////////////////
-  lazy val trips: Vector[Trip] = tours.flatMap(_.trips)
-  lazy val activities: Vector[Activity] = tours.flatMap(_.trips.map(_.activity))
-  lazy val legs: Vector[Leg] = tours.flatMap(_.trips.map(_.leg)).flatten
+  lazy val trips: Array[Trip] = tours.flatMap(_.trips)
+  lazy val activities: Array[Activity] = tours.flatMap(_.trips.map(_.activity))
+  lazy val legs: Array[Leg] = tours.flatMap(_.trips.map(_.leg)).flatten
   private val actsLegToTrip: mutable.Map[PlanElement, Trip] = mutable.Map()
 
-  private val strategies: mutable.Map[PlanElement, mutable.Map[Class[_ <: Strategy], Strategy]] =
-    mutable.Map()
   // Beam-Specific members
-  var tours: Vector[Tour] = Vector()
+  var tours: Array[Tour] = Array()
   // Implementation of Legacy Interface
   private var person: Person = _
   private var actsLegs: Vector[PlanElement] = Vector()
   private var score: Double = Double.NaN
   private var planType: String = ""
 
   def createToursFromMatsimPlan(): Unit = {
-    tours = Vector()
+    val toursBuilder = mutable.ArrayBuilder.make[Tour]()
     var nextTour = new Tour
     var nextLeg: Option[Leg] = None
     actsLegs.foreach {
       case activity: Activity =>
         val nextTrip = Trip(activity, nextLeg, nextTour)
         nextTour.addTrip(nextTrip)
         if (activity.getType.equalsIgnoreCase("home")) {
-          tours = tours :+ nextTour
+          toursBuilder += nextTour
           nextTour = new Tour
         }
       case leg: Leg =>
         nextLeg = Some(leg)
     }
-    if (nextTour.trips.nonEmpty) tours = tours :+ nextTour
+    if (nextTour.trips.nonEmpty) toursBuilder += nextTour
+    tours = toursBuilder.result()
     indexBeamPlan()
-    actsLegs.foreach {
-      case l: Leg =>
-        putStrategy(actsLegToTrip(l), ModeChoiceStrategy(BeamMode.fromString(l.getMode)))
-      case _ =>
-    }
   }
```

Comment:
Why this part was removed?

Reply:
This was about updating `strategies` map in BeamPlan. I could not find any usage of this map. 

Previously it was removed from `ChoosesMode` here https://github.com/LBNL-UCB-STI/beam/commit/343672c43ef62281d842fd041c81e6983f126d15 and then from `PersonAgent` here https://github.com/LBNL-UCB-STI/beam/commit/d99b8cf09729e9c6d08a05ce86c118a385861b18. 
 
So we only inserting values there and don't use them. That is why I dropped this variable and related code for updates. Actually it looks that `beam.agentsim.agents.planning.Strategy` could be dropped as well after that.

Reply:
It seems it is used only in test, looks like some forgotten old code, ok then, let's remove it